File: deaf_grandma.js
function deafGrandma() {

    let goodbyCounter = 0
    let input = ""
    let response = ""

    // grandma start conversation
    alert("HEY KID!")

    while (goodbyCounter <= 1) { // continue talking to grandma so long as we haven't said goodbye more than once

        input = window.prompt("Respond to grandma:")

        // say nothing
        if (input === "") {
            response = "WHAT?!"
            // any lower case in input
        } else if (input.toUpperCase() !== input) {
            response = "SPEAK UP, KID!"
            // normal question in all caps
        } else if (input === "GOODBYE!") {
            goodbyCounter++

            if (goodbyCounter > 1) { // we've said goobye once before
                response = "LATER, SKATER!"
            }
            else { // this is the first time we're saying goodbye
                response = "LEAVING SO SOON?"
            }

        } else {
            response = "NO, NOT SINCE 1946!"
        }

        alert(response)
    }
}

deafGrandma();

File: deaf_grandma.py
def deaf_grandma():
    # track the number of times user says goobye
    goodbye_counter = 0

    # initiate conversation with grandma
    response = input("HEY KID!\n")

    # continue talking to grandma so long as user has not said goobye twice
    while goodbye_counter < 2:
        #increase goodbye counter for each goodbye and break the loop when it hits 2
        if response == "GOODBYE!" and goodbye_counter == 1:
                print("LATER SKATER!")
                goodbye_counter += 1
                break
        if response == "GOODBYE!" and goodbye_counter == 0:
                response = input("LEAVING SO SOON?\n")
                goodbye_counter += 1
        # if there is no response
        if response == "":
            response == input("WHAT?!\n")
        # if the response is in all uppercase
        elif response.upper() == response:
            response = input("NO, NOT SINCE 1946!\n")
        # any other conditions that include lowercase letters
        else:
            response = input("SPEAK UP KID!\n")
        
deaf_grandma()
